### PR TITLE
Improve roxygen documentation

### PR DESCRIPTION
**Pull request by RasmusSkytte**
### Intent
This PR takes a thorough look at the roxygen documentation and aligns the style internally and with the mlr3 style.
(This fixes the `param_and_field_linter()` lints we have been getting since merging PR #102)

In addition, some use of the `db_table_id` argument have been renamed to `db_table` to align names of arguments across functions.
The recent updates to `id()`, means that there we no longer need to supply an `Id` but just something coercible by `id()`.

There also was some imports libraries that was causing issues. I removed these from imports and instead ensured every function call was visibly binded.
This also meant we could have fewer dependencies and move some dependencies to "Suggests" and not rely on these.

### Approach
* Argument `db_table_id` renamed to `db_table`.

Roxygen docs:
* Datatypes of arguments have been made explicit by adding the specification ```(`<datatype>()`)\cr``` to all function args.

* The indentation of roxygen docs have been aligned across functions

* The order of roxygen tags have been aligned (but is not actively enforced)

	* title
	* description
	* param
	* return
	* examples
	* imports
	* export

* Punctuation marks added throughout.

* Examples made conditional on the installation status of `RSQLite`.
  This mean that with a few more changes, we could move `RSQLite` to "Suggests" and don't have it as a dependency.

### Known issues
N/A

### Checklist

* [x] The PR passes all local unit tests
* [x] I have documented any new features introduced
* [x] If the PR adds a new feature, please add an entry in `NEWS.md`
* [x] A reviewer is assigned to this PR

### CODE REVIEW

File: R/digest_to_checksum.R [hunk 1]
```diff
@@ -7,12 +7,14 @@
...
 #' @template .data
-#' @param col Name of the column to put the checksums in
-#' @param exclude Columns to exclude from the checksum generation
+#' @param col (`character(1)`)\cr
+#'   Name of the column to put the checksums in. Will be generated if missing.
+#' @param exclude (`character()`)\cr
+#'   Columns to exclude from the checksum generation.
+#' @return
+#'   .data with an checksum column added.
 #' @examples
```
Comment (LasseEngboChr) [inline R/digest_to_checksum.R:16]: suggestion
```
#'   .data with a checksum column added.
```

File: R/get_tables.R [hunk 2]
```diff
@@ -1,21 +1,22 @@
...
 #'
 #' @template conn
-#' @param pattern A regex pattern with which to subset the returned tables
-#' @param show_temporary (`logical`)\cr
+#' @param pattern (`character(1)`)\cr
+#'   Regex pattern with which to subset the returned tables.
+#' @param show_temporary (`logical(1)`)\cr
 #'   Should temporary tables be listed?
+#' @return
+#'   A data.frame containing table names in the DB.
```
Comment (LasseEngboChr) [inline R/get_tables.R:9]: Including schema and catalog when relevant? Or information that uniquely defines all available tables?

File: R/helpers.R [hunk 3]
```diff
@@ -137,9 +142,11 @@ assert_timestamp_like <- function(timestamp, ..., add = NULL) {
 
 
 #' checkmate helper: Assert for "generic" db_table type
-#' @param db_table Object to test if is of class "tbl_dbi" or character on form "schema.table"
-#' @param ...      Parameters passed to checkmate::check_*
-#' @param add      `AssertCollection` to add assertions to
+#' @param db_table (`any(1)`)\cr
+#'   Object to test if is of class `tbl_dbi` or `character` on form "schema.table".
```
Comment (LasseEngboChr) [inline R/helpers.R:146]: schema.table => [catalog.]schema.table ?

**Review by LasseEngboChr — APPROVED** [review]
Lots of small adjustments to align the style :-)
Only a couple of suggestions.

**Comment by RasmusSkytte** [comment]
Fixes #69

**Pull request merged**